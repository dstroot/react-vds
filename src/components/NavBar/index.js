import React from 'react';
import { Link } from '@reach/router';
import logo from 'logo.png';
import * as routes from '../../constants/routes';

// isCurrent - true if the location.pathname is exactly the same as the anchor’s href.
const isActive = ({ isCurrent }) => {
  return isCurrent
    ? { className: 'active nav-link' }
    : { className: 'nav-link' };
};

// `getProps` Calls up to you to get props for the underlying anchor element.
// Useful for styling the anchor as active.
const ExactNavLink = props => <Link getProps={isActive} {...props} />;

// As you may already know, a common way to tune a React component for performance
// is to make it “pure,” which causes it to only re-render when its props change
// (instead of every time its parent re-renders). This can be done automatically
// by extending React.PureComponent instead of React.Component.

// In this case I tried PureComponent and found that the Navbar didn't
// update for all navigation activities so I went back to using a standard
// component.

class NavBar extends React.PureComponent {
  buttonRef = React.createRef();
  state = {
    isHidden: true,
  };

  toggleHidden = event => {
    this.setState({
      isHidden: !this.state.isHidden,
    });
    // blur the button (otherwise it stays pressed)
    this.buttonRef.current.blur();
  };

  render() {
    // hide and show classes
    let buttonClasses = 'navbar-toggler';
    if (this.state.isHidden) {
      buttonClasses = 'navbar-toggler collapsed';
    }
    let dropDown = 'navbar-collapse collapse show';
    if (this.state.isHidden) {
      dropDown = 'navbar-collapse collapse';
    }

    return (
      <nav className="navbar fixed-top navbar-expand-sm navbar-light bg-light">
        <div className="container">
          <Link to="/" className="navbar-brand">
            <img
              className="d-inline-block align-top"
              src={logo}
              width="30"
              height="30"
              alt=""
            />{' '}
            Verite Data Science
          </Link>

          <button
            className={buttonClasses}
            type="button"
            aria-label="Toggle navigation"
            onClick={this.toggleHidden}
            ref={this.buttonRef} // get a reference to the button
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className={dropDown}>
            <ExactNavLink to={routes.ABOUT} onClick={this.toggleHidden}>
              About
            </ExactNavLink>

            <ExactNavLink to={routes.CONTACT} onClick={this.toggleHidden}>
              Contact
            </ExactNavLink>
          </div>
        </div>
      </nav>
    );
  }
}

export default NavBar;
