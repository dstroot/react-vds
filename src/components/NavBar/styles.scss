//
// Main navbar
//

.navbar {
  box-shadow: 0 2px 40px 0 rgba(0, 0, 0, 0.1);
  text-transform: uppercase;
  a {
    text-transform: uppercase;
    font-weight: 400;
    // color: theme-color-level('primary', -8);
    color: gray('500');
    &:hover {
      color: theme-color('primary');
    }
  }
  a.active {
    color: theme-color('primary');
  }
}
