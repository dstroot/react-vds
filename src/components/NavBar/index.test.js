import React from 'react';
import renderer from 'react-test-renderer';

import NavBar from '../NavBar';

test('NavBar renders a Navbar', () => {
  const component = renderer.create(<NavBar />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
