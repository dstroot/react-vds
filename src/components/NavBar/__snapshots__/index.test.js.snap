// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders 1`] = `
<DocumentFragment>
  <nav
    class="navbar fixed-top navbar-expand-sm navbar-light bg-light"
  >
    <div
      class="container"
    >
      <a
        aria-current="page"
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          class="d-inline-block align-top"
          height="30"
          src="logo.png"
          width="30"
        />
         Verite Data Science
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <a
          class="nav-link"
          href="/about"
        >
          About
        </a>
        <a
          class="nav-link"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
